### PR TITLE
implement digest auth for netcams

### PR DESCRIPTION
**Pull request by jamesshoemaker**

### CODE REVIEW

**Comment by jamesshoemaker** [comment]
It's a little hacky, but it works

**Comment by jamesshoemaker** [comment]
any suggestions on making it less hacky?

File: netcam.h [hunk 1]
```diff
@@ -118,6 +118,13 @@ struct netcam_caps {                    /* netcam capabilities: */
         unsigned char content_length;   /*  0 - unsupported     */
 } caps;
 
+typedef struct 
+{
+	char server_realm[255];
```
Comment (tosiara) [inline netcam.h]: Is 255 hardcoded by RFC?

File: netcam_http.c [hunk 2]
```diff
@@ -334,282 +441,289 @@ int netcam_read_next_header(netcam_context_ptr netcam)
...
+								 ,_("Error reading first header (%s)"), header);
+			free(header);
+			return -1;
+		}
+				
+		if (firstflag) {
+			if ((ret = http_result_code(header)) != 200) {
+				MOTION_LOG(DBG, TYPE_NETCAM, NO_ERRNO,_("HTTP Result code %d"), ret);
+
+				if(ret==401) //try digest auth
```
Comment (tosiara) [inline netcam_http.c]: Shouldn't we rely on the server's responce indicating "WWW-Authenticate: Digest"?

**Comment by tosiara** [comment]
One significant thing to consider is code style. You could help the reviewers by avoiding unnecessary whitespace changes and following current style of space indents and brackets position

**Comment by Mr-DaveDev** [comment]
Looks like spaces were converted to tabs in the file.  Would need to undo that revision in order to merge.  Otherwise we will have to close PR.

**Comment by Mr-Dave** [comment]
Closing.  

The use of the prefix `mjpeg` and the ffmpeg libraries as described in the [documentation](https://motion-project.github.io/motion_config.html#netcam_url) will allow users to have digest authentication.

**Pull request closed without merging**